src/main.ts

```diff
@@ -243,12 +245,15 @@ export function validateSelfSignedCertificate(
 }
 
 /**
- * Sign a string with an RSA private key and verify that the signature is valid for the RSA
- * public key in the certificate.
+ * Sign a SHA256 hash of the provided string with an RSA private key and verify that the signature
+ * is valid for the RSA public key in the certificate. The verification part is most useful for
+ * debugging, so while this may be used in server implementation for expo-updates code signing,
+ * a similar method without verification can be created for efficiency for use in production.
+ *
  * @param privateKey RSA private key
  * @param certificate X.509 certificate
- * @param stringToSign string for which to generate a signature and verify
- * @returns base64-encoded signature
+ * @param stringToSign string for which to hash, generate a signature and verify
```

Comment:
the "for which" applies only to "generate a signature"

```suggestion
 * @param stringToSign string for which to hash and generate a signature and verify
```
```suggestion
 * @param stringToSign string for which to hash, generate a signature and verify
```